Jenkinsfile

```diff
@@ -0,0 +1,227 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2020 Joyent, Inc.
+ */
+
+@Library('jenkins-joylib@v1.0.4') _
+
+pipeline {
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '30'))
+        timestamps()
+    }
+    agent none
+
+    // build once per day
+    triggers {
+        cron('H H * * *')
```

Comment:
This is the same as currently, which is all good. I wonder if there is value in scheduling this to complete a little before nightly reflashes kick off? Which are:

- nightly-1, 2, 3:

```
# Early in the morning PDT
H 4 * * *
```

- nightly-4:

```
# Early in the morning PDT
H 6 * * *
```

Current builds are ~25 minutes (https://jenkins.joyent.us/job/headnode/buildTimeTrend), so setting this to something like:

```
// A couple hours before nightly reflashes tend to kick off, at 'H 4 * * *',
// so we get reasonably up to date headnode images. 
cron('H 2 * * *')
```


Thoughts?  Could be a separate change as well.

Reply:
Yep, that's sensible - I'll fold that into this change, thanks.